### PR TITLE
AVX optimized implementation of accumulate functions migrated

### PR DESCRIPTION
**Pull request by terfendail**
**WIP**
### This pullrequest changes
Migrated accumulate AVX optimization to separate file

### CODE REVIEW

File: modules/imgproc/src/accum.cpp [hunk 1]
```diff
@@ -1493,10 +1182,24 @@ struct AccW_SIMD<ushort, double>
...
+template <typename T, typename AT> int AccProd_AVX(const T *, const T *, AT *, const uchar *, int, int);
+template <typename T, typename AT> int AccW_AVX(const T *, AT *, const uchar *, int, int, AT);
+#endif
+
 template<typename T, typename AT> void
 acc_( const T* src, AT* dst, const uchar* mask, int len, int cn )
 {
-    int i = Acc_SIMD<T, AT>()(src, dst, mask, len, cn);
+    int i;
+
```
Comment (mshabunin) [inline modules/imgproc/src/accum.cpp]: Doc builder produces warning here:
```
modules/imgproc/src/accum.cpp:1196: trailing whitespace.
+    
```

File: modules/imgproc/src/accum.cpp [hunk 2]
```diff
@@ -1493,10 +1182,24 @@ struct AccW_SIMD<ushort, double>
...
 template<typename T, typename AT> void
 acc_( const T* src, AT* dst, const uchar* mask, int len, int cn )
 {
-    int i = Acc_SIMD<T, AT>()(src, dst, mask, len, cn);
+    int i;
+    
+#if CV_AVX
+    if (checkHardwareSupport(CPU_AVX))
+        i = Acc_AVX<T, AT>(src, dst, mask, len, cn);
```
Comment (alalek) [inline modules/imgproc/src/accum.cpp]: Please rebase and make changes similar to this patch:
https://github.com/opencv/opencv/pull/9007/files

Currently "dispatching" doesn't work at all.

**Comment by tomoaki0705** [comment]
@terfendail , could you have a look on #9024 , please ?
I tried to add dispatch mode for these accumulate function series, and my patch will make a conflict with yours.  I really didn't do this on purpose, but I thought I should notice you.

I'm OK to drop my PR, but since we want to add a dispatch mode for each optimization, it might be worth to add SSE2/AVX/AVX2/NEON at the same time. Still, my PR is not yet finished.
I just wanted to know your opinion ( and from others ) which way to go (yours or mine or other choice), before either (or both) of us waste some time writing duplicated codes.

Again, I really didn't do this by any thought to heart you or over write your patch.

**Comment by terfendail** [comment]
@tomoaki0705 , your [patch](https://github.com/opencv/opencv/pull/9024) looks more consistent and IMO is a better way to go. So I'll suspend my work on accumulate functions.

**Comment by alalek** [comment]
#9024

**Pull request closed without merging**